weed/replication/sub/notification_gocdk_pub_sub.go

```diff
@@ -0,0 +1,50 @@
+package sub
+
+import (
+	"context"
+
+	"github.com/chrislusf/seaweedfs/weed/glog"
+	"github.com/chrislusf/seaweedfs/weed/pb/filer_pb"
+	"github.com/chrislusf/seaweedfs/weed/util"
+	"github.com/golang/protobuf/proto"
+	"gocloud.dev/pubsub"
+	_ "gocloud.dev/pubsub/awssnssqs"
+	_ "gocloud.dev/pubsub/azuresb"
+	_ "gocloud.dev/pubsub/gcppubsub"
+	_ "gocloud.dev/pubsub/natspubsub"
+	_ "gocloud.dev/pubsub/rabbitpubsub"
+)
+
+func init() {
+	NotificationInputs = append(NotificationInputs, &GoCDKPubSubInput{})
+}
+
+type GoCDKPubSubInput struct {
+	sub *pubsub.Subscription
+}
+
+func (k *GoCDKPubSubInput) GetName() string {
+	return "gocdk_pub_sub"
+}
+
+func (k *GoCDKPubSubInput) Initialize(config util.Configuration) error {
+	subURL := config.GetString("sub_url")
```

Comment:
should this be "topic_url"?

Reply:
No, for receiving we want a subscription.

There was a typo in the log message. Fixed.

Reply:
seems this is different from existing "notification.gocdk_pub_sub.topic_url". Is a "notification.gocdk_pub_sub.sub_url" needed in the scaffold?

https://github.com/chrislusf/seaweedfs/blob/master/weed/command/scaffold.go#L192
https://github.com/chrislusf/seaweedfs/blob/master/weed/notification/gocdk_pub_sub/gocdk_pub_sub.go#L47

Reply:
Thanks for noticing. I added an entry.

Reply:
Actually, this also need to change? I suppose the topic_url and sub_url are referring to the same thing.

https://github.com/chrislusf/seaweedfs/blob/master/weed/notification/gocdk_pub_sub/gocdk_pub_sub.go#L47

Reply:
They aren't referring to the same thing. In many Pub/Sub systems (Google Pub/Sub, Azure Service Bus, etc.) there are two separate concepts: topics and subscriptions. You publish to a topic. You receive from a subscription. They have different names, and usually must be created and deleted separately. When you create a subscription, you name the topic that it receives from, but other than that they are independent resources.